compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3250,11 +3258,7 @@ private BUnionType createNextReturnType(Location pos, BStreamType streamType) {
 
         LinkedHashSet<BType> retTypeMembers = new LinkedHashSet<>();
         retTypeMembers.add(recordType);
-        List<BType> errorTypes = getTypesList(streamType.error);
-        errorTypes.removeIf(type -> type.tag == TypeTags.NEVER);
-        if (!errorTypes.isEmpty()) {
-            retTypeMembers.addAll(errorTypes);
-        }
+        retTypeMembers.addAll(types.getAllTypes(streamType.error));
```

Comment:
This `streamType.error`  can be a `()` or a union with `()` right? If so, after executing the following line, there will be 2 `()`s get added to the union type right?

Reply:
since it's a HashSet, there won't be duplicates